cli/kata-check.go

```diff
@@ -454,3 +459,45 @@ func genericCheckKVMExtensions(extensions map[string]kvmExtension) (map[string]i
 
 	return results, nil
 }
+
+// func checkVersionConsistencyInComponents checks version consistency in kata components.
+func checkVersionConsistencyInComponents(c *cli.Context) error {
+	config, ok := c.App.Metadata["runtimeConfig"].(oci.RuntimeConfig)
+	if !ok {
+		return errors.New("cannot determine runtime config")
+	}
+
+	proxyInfo, _ := getProxyInfo(config)
+
+	shimInfo, err := getShimInfo(config)
+	if err != nil {
+		return err
+	}
+
+	var runtimeVersion VersionInfo
+	versions := strings.Split(version, ".")
```

Comment:
This works but it may be better to  use the already-vendored semver package (`github.com/blang/semver`).